scrapy_onedrive_exporter/onedrive_exporter.py

```diff
@@ -0,0 +1,58 @@
+import logging
+
+import requests
+from scrapy.exceptions import NotConfigured
+from scrapy.extensions.feedexport import BlockingFeedStorage, build_storage
+
+logger = logging.getLogger(__name__)
+
+GRAPH_API_ENDPOINT = "https://graph.microsoft.com/v1.0"
+
+
+class OneDriveFeedStorage(BlockingFeedStorage):
+    def __init__(self, uri, access_token=None, *, feed_options=None):
+        if not access_token:
+            raise NotConfigured("Missing ONEDRIVE_ACCESS_TOKEN")
+
+        if feed_options and feed_options.get("overwrite", True) is False:
+            raise NotConfigured(
+                "OneDriveFeedStorage does not support appending to files. Please "
+                "remove the overwrite option from your FEEDS setting or set it to True."
+            )
+
+        self.session_headers = {
+            'Authorization': f'Bearer {access_token}',
+            'Content-Type': 'application/json'
+        }
+        self.file_path = uri.replace('onedrive://', '')
```

Comment:
Let’s not assume which scheme the user chose (e.g. maybe they went for `od`).

Reply:
@Gallaecio the documentation mentions the scheme should be onedrive, but I updated to handle others as well. Let me know